mypy/types.py

```diff
@@ -1486,112 +1486,112 @@ def keywords_str(self, a: Iterable[Tuple[str, Type]]) -> str:
         ])
 
 
-# These constants define the method used by TypeQuery to combine multiple
-# query results, e.g. for tuple types. The strategy is not used for empty
-# result lists; in that case the default value takes precedence.
-ANY_TYPE_STRATEGY = 0   # Return True if any of the results are True.
-ALL_TYPES_STRATEGY = 1  # Return True if all of the results are True.
+# Combination strategies for boolean type queries
+def ANY_TYPE_STRATEGY(current: bool, accumulated: bool) -> bool:
+    """True if any type's result is True"""
+    if accumulated:
+        raise ShortCircuitQuery()
+    return current
```

Comment:
Compare `StopIteration` for the paradigm I am trying to use.  I thought about using `raise StopIteration` literally, but since this is not exactly an iterator I made my own specialized exception.

(In other words, I like it this way, but I would be open to the idea of using `StopIteration` instead of my custom thing if that seems cleaner)